projects/step-frontend/src/lib/modules/execution/components/execution-tabs/execution-tabs.component.ts

```diff
@@ -1,18 +1,72 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
-import { ExecutionTab } from '../../shared/execution-tab';
 import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
-import { AJS_MODULE } from '@exense/step-core';
+import { AJS_MODULE, Dashlet, Tab, ViewRegistryService } from '@exense/step-core';
+import {
+  ExecutionTabsCommand,
+  ExecutionsTabsControlService,
+  ExecutionTabsCommandType,
+} from '../../services/executions-tabs-control.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'step-execution-tabs',
   templateUrl: './execution-tabs.component.html',
   styleUrls: ['./execution-tabs.component.scss'],
 })
 export class ExecutionTabsComponent {
-  @Input() tabs: ExecutionTab[] = [];
+  @Input() executionId?: string;
   @Input() activeTabId?: string;
   @Output() activeTabIdChange = new EventEmitter<string>();
-  @Output() closeTab = new EventEmitter<string>();
+
+  readonly tabs: Array<Tab>;
+
+  private subscription: Subscription;
+
+  constructor(
+    private _viewRegistryService: ViewRegistryService,
+    private _executionTabsControlService: ExecutionsTabsControlService
+  ) {
+    this.tabs = this._viewRegistryService
+      .getDashlets('executionTab')
+      .filter(function (dashlet) {
+        return dashlet.isEnabledFct ? dashlet.isEnabledFct() : false;
+      })
+      .map<Tab>((dashlet) => this.dashletToTab(dashlet));
```

Comment:
In this case I don't agree. It's maybe a matter of coding style. Instead of optimizing towards as few lines of code as possible, I tend to optimize towards readability. And I think that this line:
`.map<Tab>((dashlet) => this.dashletToTab(dashlet));` is more readable than that:
`.map<Tab>(({id, label}) => ({id, label, disabled: false}))`

Reply:
I think that methods are useful not only to avoid redundancy, but also to give pieces of code a name and explain what the code is doing to a human reader.